### PR TITLE
feat(array-exercise04): add array exercise04

### PR DESCRIPTION
**Pull request by ThuongHoai081**
4. Viết chương trình nhận vào 2 mảng các số nguyên đã sắp xếp, tạo mảng số nguyên từ 2 mảng số nguyên đã cho và giữ nguyên thứ tự đã sắp xếp. (Yêu cầu: không dùng thêm thao tác sort)

### CODE REVIEW

File: src/main/java/array/Exercise04.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        int j = 0;
+        int k = 0;
+
+        while (i < len1 && j < len2) {
+            if (arr1[i] <= arr2[j]) {
+                mergedArray[k++] = arr1[i++];
+                continue;
+            }
+
+            mergedArray[k++] = arr2[j++];
```
Comment (nguyenvanman) [inline src/main/java/array/Exercise04.java:27]:
```suggestion
            if (arr1[i] <= arr2[j]) {
                mergedArray[k++] = arr1[i++];
            } else {
                 mergedArray[k++] = arr2[j++];
             }
```
I think using an else block makes the code easier to read and understand in this case compared to using continue.

File: src/test/java/array/Exercise04Test.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    @Test
+    void shouldMergeTwoSortedArrays_OK() {
+        final Exercise04 ex = new Exercise04();
+        assertArrayEquals(new int[]{1, 2, 3, 4, 5, 6}, ex.mergeArrayAndSort(new int[]{1, 3, 5}, new int[]{2, 4, 6}));
+        assertArrayEquals(new int[]{1, 2, 3}, ex.mergeArrayAndSort(new int[]{1, 2, 3}, new int[]{}));
+        assertArrayEquals(new int[]{1, 2, 3}, ex.mergeArrayAndSort(new int[]{}, new int[]{1, 2, 3}));
+        assertArrayEquals(new int[]{}, ex.mergeArrayAndSort(new int[]{}, new int[]{}));
+        assertArrayEquals(new int[]{1, 2, 3}, ex.mergeArrayAndSort(null, new int[]{1, 2, 3}));
+        assertArrayEquals(new int[]{1, 2, 3}, ex.mergeArrayAndSort(new int[]{1, 2, 3}, null));
+        assertArrayEquals(null, ex.mergeArrayAndSort(null, null));
```
Comment (nguyenvanman) [inline src/test/java/array/Exercise04Test.java:17]: Có một trường hợp kiểm tra bị thiếu khi arr1 hoặc arr2 là null.

Comment (nguyenvanman) [inline src/test/java/array/Exercise04Test.java:17]: There’s a missing test case for when either arr1 or arr2 is null.

Comment (ThuongHoai081) [reply src/test/java/array/Exercise04Test.java:17]: I think I already covered that case in my tests at lines 15 and 16, where I checked when either arr1 or arr2 is null.
